File: src/main/java/kr/co/bookand/backend/config/jwt/RefreshToken.java
package kr.co.bookand.backend.config.jwt;

import kr.co.bookand.backend.account.domain.Account;
import lombok.*;

import javax.persistence.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "refresh_token")
@Entity
@Builder
public class RefreshToken {

    @Id
    @Column(name = "rt_key")
    private String key;

    @Column(name = "rt_value")
    private String value;

    @OneToOne(fetch = FetchType.LAZY,  cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "account_id")
    private Account account;
}

File: src/main/kotlin/kr/co/bookand/backend/config/jwt/RefreshToken.kt
package kr.co.bookand.backend.config.jwt

import kr.co.bookand.backend.account.model.Account
import javax.persistence.*

@Entity
@Table(name = "refresh_token")
class RefreshToken (
    @Id
    @Column(name = "rt_key")
    var key: String,

    @Column(name = "rt_value")
    var value: String,

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    var account: Account
){
    fun updateValue(value: String) {
        this.value = value
    }
}